daml-lf/transaction-scalacheck/src/main/scala/com/digitalasset/daml/lf/value/ValueGenerators.scala

```diff
@@ -196,6 +196,13 @@ object ValueGenerators {
     } yield ValueTextMap[ContractId](SortedLookupList(Map(list: _*)))
   def valueMapGen: Gen[ValueTextMap[ContractId]] = valueMapGen(0)
 
+  private def valueGenMapGen(nesting: Int) =
+    Gen
+      .listOf(Gen.zip(Gen.lzy(valueGen(nesting)), Gen.lzy(valueGen(nesting))))
+      .flatMap(list => ValueGenMap[ContractId](ImmArray(list)))
```

Comment:
```suggestion
      .map(list => ValueGenMap[ContractId](ImmArray(list)))
```

I'll leave you to 😨 😢 about why it compiled